warehouse/observations/models.py

```diff
@@ -244,6 +249,7 @@ def record_observation(
             observer=actor.observer,
             payload=payload,
             related=self,
```

Comment:
feels bad to be implicitly binding to `name`, should we perhaps use an explicit `related_name` property for models we may denormalize as such?

Reply:
I tried to use the pre-exisitng `__repr__()` call that these models implement, but it didn't exactly look right.

The resulting strings look like: `"Project(name='example')"` for a Project, and `"Release(project=Project(name='example'), version='0.0.2')"`.

These are semi-seralized, as in we could feasibly turn them back into structured data, but they likely won't hold up to efficient SQL queries - but maybe they don't have to?

Reply:
I've updated the scope to use `repr()` for these, and dropped the bind to `name`.